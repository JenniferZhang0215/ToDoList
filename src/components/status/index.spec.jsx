import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Status from "./index";

const props = {
  statusArray: ["All", "Processing", "Done"],
  onStatus: jest.fn(),
  selectedStatus: "All",
};

configure({ adapter: new Adapter() });
describe("<Status />", () => {
  const wrapper = shallow(<Status {...props} />);

  it("should render default snapshot", () => {
    expect(wrapper).toMatchSnapshot("default");
  });

  describe("span", () => {
    const status = "Processing";
    const span = wrapper
      .findWhere((node) => node.key() === status)
      .find("span");
    it("should call onStatus if status text is clicked", () => {
      span.simulate("click");
      expect(props.onStatus).toHaveBeenCalledWith(status);
    });

    it("should render status text correctly", () => {
      expect(span.text()).toEqual(status);
    });
  });
});
