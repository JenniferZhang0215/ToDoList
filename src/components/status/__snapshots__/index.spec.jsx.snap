// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render default snapshot: default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status
    onStatus={[MockFunction]}
    selectedStatus="All"
    statusArray={
      Array [
        "All",
        "Processing",
        "Done",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="navbar-nav"
      >
        <li
          className="nav-item active"
        >
          <span
            className="nav-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            All
          </span>
        </li>
        <li
          className="nav-item "
        >
          <span
            className="nav-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            Processing
          </span>
        </li>
        <li
          className="nav-item "
        >
          <span
            className="nav-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            Done
          </span>
        </li>
      </ul>,
      "className": "navbar navbar-expand-lg navbar-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item active"
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              All
            </span>
          </li>,
          <li
            className="nav-item "
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Processing
            </span>
          </li>,
          <li
            className="nav-item "
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Done
            </span>
          </li>,
        ],
        "className": "navbar-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "All",
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              All
            </span>,
            "className": "nav-item active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All",
              "className": "nav-link",
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
            },
            "ref": null,
            "rendered": "All",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "Processing",
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Processing
            </span>,
            "className": "nav-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Processing",
              "className": "nav-link",
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
            },
            "ref": null,
            "rendered": "Processing",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "Done",
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Done
            </span>,
            "className": "nav-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Done",
              "className": "nav-link",
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
            },
            "ref": null,
            "rendered": "Done",
            "type": "span",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="navbar-nav"
        >
          <li
            className="nav-item active"
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              All
            </span>
          </li>
          <li
            className="nav-item "
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Processing
            </span>
          </li>
          <li
            className="nav-item "
          >
            <span
              className="nav-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Done
            </span>
          </li>
        </ul>,
        "className": "navbar navbar-expand-lg navbar-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="nav-item active"
            >
              <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                All
              </span>
            </li>,
            <li
              className="nav-item "
            >
              <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                Processing
              </span>
            </li>,
            <li
              className="nav-item "
            >
              <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                Done
              </span>
            </li>,
          ],
          "className": "navbar-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "All",
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                All
              </span>,
              "className": "nav-item active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "All",
                "className": "nav-link",
                "onClick": [Function],
                "style": Object {
                  "cursor": "pointer",
                },
              },
              "ref": null,
              "rendered": "All",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Processing",
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                Processing
              </span>,
              "className": "nav-item ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Processing",
                "className": "nav-link",
                "onClick": [Function],
                "style": Object {
                  "cursor": "pointer",
                },
              },
              "ref": null,
              "rendered": "Processing",
              "type": "span",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Done",
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                Done
              </span>,
              "className": "nav-item ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Done",
                "className": "nav-link",
                "onClick": [Function],
                "style": Object {
                  "cursor": "pointer",
                },
              },
              "ref": null,
              "rendered": "Done",
              "type": "span",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
